src/components/30-organisms/footer/index.js

```diff
@@ -49,6 +58,77 @@ class AXAFooter extends LitElement {
     this.clickevents = false;
     this.onItemClick = () => {};
     this._accordionActiveIndex = -1;
+    this.firstRender = true;
+  }
+
+  firstUpdated() {
+    this.inlineStyles('resetHeadingCss');
+  }
+
+  prepareSlotsWithIndexes() {
+    // set slotted numbers automatically for all titles and all items for each column
+    // every title followed with items will be a column (DOM sequence defines order)
+    // There is no wrapper for the columns but a counter that controls which
+    // column to use
+    // EXAMPLE IN PSEUDO DOM:
+    //
+    // TITLE -> start column 0
+    // ITEM -> add item to column 0
+    // ITEM -> add item to column 0
+    // ITEM -> add item to column 0
+    // ITEM -> add item to column 0
+    //
+    // TITLE -> start column 1
+    // ITEM -> add item to column 1
+    // ITEM -> add item to column 1
+    // ITEM -> add item to column 1
+
+    const childrenArr = Array.prototype.slice.call(this.children);
+
+    const filter = criteria => child =>
+      child.getAttribute('slot').includes(criteria);
+
+    const noHeaderFilter = criteria => child => {
+      const { nodeName } = child;
+      return (
+        filter(criteria)(child) && !HEADINGS.includes(nodeName.toLowerCase())
+      );
+    };
+
+    const onlyColumns = childrenArr.filter(
+      // only accepts those slots that are columns
+      filter('column-')
+    );
+    const onlySocials = childrenArr.filter(
+      // only accepts those slots that are social columns
+      noHeaderFilter('social-')
+    );
+
+    let currentColumnIndex = -1;
+    let totalAmountPreviousColumns = 0;
+    onlyColumns.forEach((child, index) => {
+      const { nodeName } = child;
+      if (HEADINGS.includes(nodeName.toLowerCase())) {
+        currentColumnIndex += 1;
+        totalAmountPreviousColumns = index;
+        child.setAttribute(
+          'slot',
+          child.getAttribute('slot').replace('-x-', `-${currentColumnIndex}-`)
+        );
+      } else {
+        // -1 cause heading is always first of index
```

Comment:
comment is difficult to comprehend - rework a bit!

Reply:
done